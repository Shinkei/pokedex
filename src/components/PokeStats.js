import React from 'react'
import { getPokemon } from '../services/pokedex'

class PokeStats extends React.Component {
  constructor () {
    super()
    this.state = { pokemon: {} }
  }

  async componentDidMount () {
    const { match: { params: { name } } = {} } = this.props
    const pokemon = await getPokemon(name)
    this.setState({ pokemon })
  }

  render () {
    const { pokemon } = this.state
    return (
      <div>
        <div>Name: <span>{pokemon.name}</span></div>
        <img src={pokemon.sprites && pokemon.sprites.front_default} alt='sprite' />
      </div>
    )
  }
}

export default PokeStats
