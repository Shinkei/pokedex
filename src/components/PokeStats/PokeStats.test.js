import React from "react";
import { render } from "@testing-library/react";
import PokeStats from "./PokeStats";

jest.mock("../../services/pokedex.js");

describe("PokeStats", () => {
  /**
   * React snapshot test example
   * The snapshot test is a general test that will save the rendered component and eveytime
   * the component is rendered again, is compared with the snapshot, if it is not exatly as
   * the snapshot, the test fails.
   */
  it("Should render Bulbasaur component", async () => {
    const pokestats = render(<PokeStats pokemon="bulbasaur" />);
    const { waitForNextUpdate, getByText } = pokestats;

    // Necessary for the component to render after fetching API
    await waitForNextUpdate;

    expect(pokestats).toMatchSnapshot();
    const nameElement = getByText("bulbasaur");
    expect(nameElement).toBeInTheDocument();

    const greenPowerElement = getByText("🍃");
    expect(greenPowerElement).toBeInTheDocument();
  });

  /**
   * React Unit test example
   * the testing library helps to render the components and then, the test gets the img element
   * and check if the element is inside the rendered component
   */
  it("should render an image", () => {
    const { getByRole } = render(<PokeStats pokemon="bulbasaur" />);
    const pokemonImage = getByRole("img");
    expect(pokemonImage).toBeInTheDocument();
  });
});
