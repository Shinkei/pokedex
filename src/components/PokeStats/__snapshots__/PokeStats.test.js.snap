// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PokeStats Should render Bulbasaur component 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div>
        <div
          class="nes-table-responsive"
          style="display: flex; justify-content: center;"
        >
          <table
            class="nes-table is-bordered"
          >
            <tbody>
              <tr>
                <td>
                  Name
                </td>
                <td>
                  bulbasaur
                </td>
                <td
                  rowspan="5"
                  style="border-bottom-width: 0; border-left-width: 4px;"
                >
                  <img
                    src="https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/1.png"
                  />
                </td>
              </tr>
              <tr>
                <td>
                  Abilities
                </td>
                <td>
                  <div>
                    chlorophyll
                  </div>
                  <div>
                    overgrow
                  </div>
                </td>
              </tr>
              <tr>
                <td>
                  Height
                </td>
                <td>
                  7
                  m
                </td>
              </tr>
              <tr>
                <td>
                  Types
                </td>
                <td>
                  <div
                    class="power-emojis"
                  >
                    üíÄ
                  </div>
                  <div
                    class="power-emojis"
                  >
                    üçÉ
                  </div>
                </td>
              </tr>
              <tr>
                <td>
                  Weight
                </td>
                <td>
                  69
                  kg
                </td>
              </tr>
            </tbody>
          </table>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div>
      <div
        class="nes-table-responsive"
        style="display: flex; justify-content: center;"
      >
        <table
          class="nes-table is-bordered"
        >
          <tbody>
            <tr>
              <td>
                Name
              </td>
              <td>
                bulbasaur
              </td>
              <td
                rowspan="5"
                style="border-bottom-width: 0; border-left-width: 4px;"
              >
                <img
                  src="https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/1.png"
                />
              </td>
            </tr>
            <tr>
              <td>
                Abilities
              </td>
              <td>
                <div>
                  chlorophyll
                </div>
                <div>
                  overgrow
                </div>
              </td>
            </tr>
            <tr>
              <td>
                Height
              </td>
              <td>
                7
                m
              </td>
            </tr>
            <tr>
              <td>
                Types
              </td>
              <td>
                <div
                  class="power-emojis"
                >
                  üíÄ
                </div>
                <div
                  class="power-emojis"
                >
                  üçÉ
                </div>
              </td>
            </tr>
            <tr>
              <td>
                Weight
              </td>
              <td>
                69
                kg
              </td>
            </tr>
          </tbody>
        </table>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
